src/nbl/asset/utils/CMeshManipulator.cpp

```diff
@@ -586,68 +624,43 @@ core::smart_refctd_ptr<ICPUMeshBuffer> IMeshManipulator::createOptimizedMeshBuff
+    outbuffer->setSkin(
+        SBufferBinding<ICPUBuffer>(reinterpret_cast<const SBufferBinding<ICPUBuffer>&>(_inbuffer->getInverseBindPoseBufferBinding())),
+        SBufferBinding<ICPUBuffer>(reinterpret_cast<const SBufferBinding<ICPUBuffer>&>(_inbuffer->getJointAABBBufferBinding())),
+        core::smart_refctd_ptr<ICPUSkeleton>(const_cast<ICPUSkeleton*>(_inbuffer->getSkeleton())),
+        _inbuffer->getMaxJointsPerVertex()
+    );
 
+    // make index buffer 0,1,2,3,4,... if nothing's mapped
 	// make 32bit index buffer if 16bit one is present
-	if (outbuffer->getIndexType() == EIT_16BIT)
-	{
-        auto ib = CMeshManipulator::create32BitFrom16BitIdxBufferSubrange(reinterpret_cast<uint16_t*>(outbuffer->getIndices()), outbuffer->getIndexCount());
-        outbuffer->setIndexBufferBinding({ 0u, std::move(ib) });
-		// no need to set index buffer offset to 0 because it already is
-		outbuffer->setIndexType(canonicalMeshBufferIndexType);
-	}
-
 	// convert index buffer for triangle primitives
-	if (outbuffer->getPipeline()->getPrimitiveAssemblyParams().primitiveType == EPT_TRIANGLE_FAN)
-	{
-        outbuffer->getPipeline()->getPrimitiveAssemblyParams().primitiveType = EPT_TRIANGLE_LIST;
-        auto indexCount = outbuffer->getIndexCount();
-		auto newIb = idxBufferFromTrianglesFanToTriangles(outbuffer->getIndices(), indexCount, canonicalMeshBufferIndexType, canonicalMeshBufferIndexType);
-		outbuffer->setIndexCount(indexCount);
-        outbuffer->setIndexBufferBinding({ 0u, std::move(newIb) });
-	}
-	else if (outbuffer->getPipeline()->getPrimitiveAssemblyParams().primitiveType == EPT_TRIANGLE_STRIP)
-	{
-        outbuffer->getPipeline()->getPrimitiveAssemblyParams().primitiveType = EPT_TRIANGLE_LIST;
-        auto indexCount = outbuffer->getIndexCount();
-		auto newIb = idxBufferFromTriangleStripsToTriangles(outbuffer->getIndices(), indexCount, canonicalMeshBufferIndexType, canonicalMeshBufferIndexType);
-		outbuffer->setIndexCount(indexCount);
-        outbuffer->setIndexBufferBinding({ 0u, std::move(newIb) });
-	}
-	else if (outbuffer->getPipeline()->getPrimitiveAssemblyParams().primitiveType != EPT_TRIANGLE_LIST)
+    constexpr auto canonicalMeshBufferIndexType = EIT_32BIT;
+    IMeshManipulator::homogenizePrimitiveTypeAndIndices(&outbuffer.get(),&outbuffer.get()+1,EPT_TRIANGLE_LIST,canonicalMeshBufferIndexType);
+    if (outbuffer->getPipeline()->getPrimitiveAssemblyParams().primitiveType != EPT_TRIANGLE_LIST)
 		return nullptr;
 
 	// STEP: weld
     createMeshBufferWelded(outbuffer.get(), _errMetric, false, false);
 
     // STEP: filter invalid triangles
-    filterInvalidTriangles(outbuffer.get());
+    if (!_inbuffer->isSkinned())
+        filterInvalidTriangles(outbuffer.get());
 
 	// STEP: overdraw optimization
-	COverdrawMeshOptimizer::createOptimized(outbuffer.get(), false);
+	COverdrawMeshOptimizer::createOptimized(outbuffer.get(),outbuffer.get());
```

Comment:
api for this has changed?

Reply:
yes, now it takes and in and out meshbuffer (no longer creates a duplicate inside)